src/server/node_manager.c

```diff
@@ -7555,7 +7560,35 @@ free_nodes(job *pjob)
 		for (j=0; j<psvrmom->msr_jbinxsz; j++) {
 			if (psvrmom->msr_jobindx[j] == pjob) {
 				psvrmom->msr_jobindx[j] = NULL;
+				if (is_called_by_job_purge)
+					special_case = 1;
+			}
+		}
+
+		if (special_case) {
+			sprintf(log_buffer, "\n================================================================"
+				"======================================================\nPBSPro diagnostic information."
+				" Share this log with PBSPro Team.\n=========================================="
+				"============================================================================\n"
+				"Mom's state:%lu, number of jobs on this node: %d, number of vnodes: %d",
+				psvrmom->msr_state, psvrmom->msr_numjobs, psvrmom->msr_numvnds);
+			log_event(PBSEVENT_DEBUG3, PBS_EVENTCLASS_JOB, LOG_INFO, pjob->ji_qs.ji_jobid, log_buffer);
+			ch = buffer;
```

Comment:
Initialise buffer to null just in case.

Reply:
Please use snprintf for log messages, since there is a good chance to exceed sizes, and also truncation of such message buffers is usually not fatal anyway.

Reply:
strcating this into one buffer is not required. Please make it simple and print one job per log line. That way, you dont have to worry about the buffer size.